src/libostree/ostree-repo-commit.c

```diff
@@ -819,7 +945,7 @@ write_object (OstreeRepo         *self,
 
   g_assert (actual_checksum != NULL); /* Pacify static analysis */
           
-  if (indexable && temp_file_is_regular)
```

Comment:
I _think_ this is intentional. Before this change, `indexable` was only set for `objtype == OSTREE_OBJECT_TYPE_FILE` and `repo_mode == OSTREE_REPO_MODE_ARCHIVE_Z2`. However, what @jhiesey was trying to achieve was getting the size for _all_ files, including metadata. That would happen in the `else` from `objtype == OSTREE_OBJECT_TYPE_FILE`.